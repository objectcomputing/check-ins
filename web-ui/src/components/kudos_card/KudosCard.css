.kudos-card .kudos-card-header {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    padding: 1rem;
    cursor: pointer;
}

.kudos-card .kudos-card-header .members-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 0.5rem;
}

.kudos-card .kudos-card-header .kudos-status-container {

}

.kudos-card .kudos-card-content {
    padding: 1rem;
}