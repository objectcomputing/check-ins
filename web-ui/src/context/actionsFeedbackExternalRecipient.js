export const SET_CSRF = '@@check-ins/update_csrf';
export const UPDATE_TOAST = '@@check-ins/update_toast';

