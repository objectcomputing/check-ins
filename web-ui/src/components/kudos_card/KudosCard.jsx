import React, { useCallback, useContext, useState } from "react";
import PropTypes from "prop-types";
import {
  Paper,
  Collapse,
  Divider,
  Typography,
  Avatar,
  Chip,
  Button,
  AvatarGroup,
  Tooltip,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions,
  TextField,
  FormGroup,
  FormControlLabel,
  Checkbox,
} from "@mui/material";
import {
  selectCsrfToken,
  selectActiveOrInactiveProfile,
} from "../../context/selectors";
import MemberSelector from '../member_selector/MemberSelector';
import { AppContext } from "../../context/AppContext";
import { getAvatarURL } from "../../api/api";
import DateFnsUtils from "@date-io/date-fns";
import CheckIcon from "@mui/icons-material/Check";
import CloseIcon from "@mui/icons-material/Close";
import EditIcon from "@mui/icons-material/Edit";
import TeamIcon from "@mui/icons-material/Groups";

import "./KudosCard.css";
import { approveKudos, deleteKudos, updateKudos } from "../../api/kudos";
import { UPDATE_TOAST } from "../../context/actions";

const dateUtils = new DateFnsUtils();

const propTypes = {
  kudos: PropTypes.shape({
    id: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    senderId: PropTypes.string.isRequired,
    recipientTeam: PropTypes.object,
    dateCreated: PropTypes.array.isRequired,
    dateApproved: PropTypes.array,
    recipientMembers: PropTypes.array,
  }).isRequired,
  includeActions: PropTypes.bool,
  includeEdit: PropTypes.bool,
  onKudosAction: PropTypes.func,
};

const KudosCard = ({ kudos, includeActions, includeEdit, onKudosAction }) => {
  const { state, dispatch } = useContext(AppContext);
  const csrf = selectCsrfToken(state);

  const [expanded, setExpanded] = useState(true);
  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
  const [editDialogOpen, setEditDialogOpen] = useState(false);
  const [kudosPublic, setKudosPublic] = useState(kudos.publiclyVisible);
  const [kudosMessage, setKudosMessage] = useState(kudos.message);
  const [memberSelectorOpen, setMemberSelectorOpen] = useState(false);
  const [kudosRecipientMembers, setKudosRecipientMembers] = useState(kudos.recipientMembers);

  const sender = selectActiveOrInactiveProfile(state, kudos.senderId);

  const getRecipientComponent = useCallback(() => {
    if (kudos.recipientTeam) {
      return (
        <Chip
          avatar={
            <Avatar>
              <TeamIcon />
            </Avatar>
          }
          label={kudos.recipientTeam.name}
        />
      );
    } else if (kudos.recipientMembers.length === 1) {
      const [recipient] = kudos.recipientMembers;
      return (
        <Chip
          avatar={<Avatar src={getAvatarURL(recipient?.workEmail)} />}
          label={`${recipient.firstName} ${recipient.lastName}`}
        />
      );
    }

    return (
      <AvatarGroup max={16}>
        {kudos.recipientMembers.map((member) => (
          <Tooltip
            arrow
            key={member.id}
            title={`${member.firstName} ${member.lastName}`}
          >
            <Avatar src={getAvatarURL(member.workEmail)} />
          </Tooltip>
        ))}
      </AvatarGroup>
    );
  }, [kudos]);

  const approveKudosCallback = useCallback(
    async (event) => {
      event.stopPropagation();
      const res = await approveKudos(kudos, csrf);
      if (res?.payload?.data && !res.error) {
        dispatch({
          type: UPDATE_TOAST,
          payload: {
            severity: "success",
            toast: "Kudos approved",
          },
        });
        onKudosAction && onKudosAction();
      } else {
        dispatch({
          type: UPDATE_TOAST,
          payload: {
            severity: "error",
            toast: "Failed to approve kudos",
          },
        });
      }
    },
    [kudos, csrf, dispatch, onKudosAction]
  );

  const deleteKudosCallback = useCallback(async () => {
    setDeleteDialogOpen(false);
    const res = await deleteKudos(kudos.id, csrf);
    if (res?.payload?.status === 204 && !res.error) {
      dispatch({
        type: UPDATE_TOAST,
        payload: {
          severity: "success",
          toast: "Pending kudos deleted",
        },
      });
      onKudosAction && onKudosAction();
    } else {
      dispatch({
        type: UPDATE_TOAST,
        payload: {
          severity: "error",
          toast: "Failed to delete kudos",
        },
      });
    }
  }, [kudos, csrf, dispatch, onKudosAction]);

  const updateKudosCallback = useCallback(async () => {
    // Close the dialog.
    setEditDialogOpen(false);

    // Update the modifiable parts.
    const proposed = {
      id: kudos.id,
      message: kudosMessage,
      publiclyVisible: kudosPublic,
      recipientMembers: kudosRecipientMembers,
    };

    // Update on the server.
    const res = await updateKudos(proposed, csrf);
    if (res.error) {
      dispatch({
        type: UPDATE_TOAST,
        payload: {
          severity: "error",
          toast: "Failed to update kudos",
        },
      });
    } else {
      dispatch({
        type: UPDATE_TOAST,
        payload: {
          severity: "success",
          toast: "Kudos Updated",
        },
      });
      onKudosAction && onKudosAction();
    }
  }, [kudos, kudosMessage, kudosPublic, kudosRecipientMembers, csrf, dispatch, onKudosAction]);

  const getStatusComponent = useCallback(() => {
    const dateApproved = kudos.dateApproved
      ? new Date(kudos.dateApproved.join("/"))
      : null;

    const info = [];
    const actions = [];
    if (includeActions) {
      actions.push(
        <Tooltip key="approve" arrow title="Approve">
          <Button
            variant="outlined"
            color="success"
            size="small"
            onClick={approveKudosCallback}
          >
            <CheckIcon />
          </Button>
        </Tooltip>
      );
    } else if (dateApproved) {
      info.push(
        <Typography key="received" color="green" variant="body2">
          Received{" "}
          {dateApproved ? dateUtils.format(dateApproved, "MM/dd/yyyy") : ""}
        </Typography>
      );
    } else {
      const dateCreated = new Date(kudos.dateCreated.join("/"));
      if (kudos.publiclyVisible) {
        info.push(
          <Typography key="pending" color="orange" variant="body2">
            Pending
          </Typography>
        );
      }
      info.push(
        <Typography key="created" variant="body2" color="gray" fontSize="10px">
          Created {dateUtils.format(dateCreated, "MM/dd/yyyy")}
        </Typography>
      );
    }
    if (includeEdit) {
      actions.push(
        <Tooltip key="edit" arrow title="Edit">
          <Button
            variant="outlined"
            size="small"
            onClick={(event) => {
              event.stopPropagation();
              reloadKudosValues();
              setEditDialogOpen(true);
            }}
          >
            <EditIcon />
          </Button>
        </Tooltip>
      );
    }
    if (includeActions || includeEdit) {
      actions.push(
        <Tooltip key="delete" arrow title="Delete">
          <Button
            variant="outlined"
            color="error"
            size="small"
            onClick={(event) => {
              event.stopPropagation();
              setDeleteDialogOpen(true);
            }}
          >
            <CloseIcon />
          </Button>
        </Tooltip>
      );
    }
    return <>
             {info.length > 0 && <div>
               {info}
             </div>}
             {actions.length > 0 && <div className="kudos-action-buttons">
               {actions}
             </div>}
           </>;
  }, [kudos, includeActions, includeEdit, approveKudosCallback]);

  const reloadKudosValues = () => {
    setKudosMessage(kudos.message);
    setKudosPublic(kudos.publiclyVisible);
    setKudosRecipientMembers(kudos.recipientMembers);
  };

  const dateApproved = kudos.dateApproved
    ? new Date(kudos.dateApproved.join("/"))
    : null;

  return (
    <>
      <Dialog open={deleteDialogOpen}>
        <DialogTitle>Delete Kudos</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Are you sure you want to complete this action? The kudos
            will be deleted.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button
            onClick={() => setDeleteDialogOpen(false)}
            style={{ color: "gray" }}
          >
            Cancel
          </Button>
          <Button onClick={deleteKudosCallback} color="error" autoFocus>
            Delete
          </Button>
        </DialogActions>
      </Dialog>
      <Dialog open={editDialogOpen}  maxWidth="sm" fullWidth>
        <DialogTitle>Edit Kudos</DialogTitle>
        <DialogContent>
          <MemberSelector
            onChange={setKudosRecipientMembers}
            selected={kudosRecipientMembers}
          />
          <FormGroup>
            <FormControlLabel
              control={<Checkbox checked={kudosPublic} />}
              label="Public"
              onChange={e => {
                setKudosPublic(e.target.checked);
              }}
            />
          </FormGroup>
          <TextField
            id="kudos-text"
            fullWidth
            multiline
            defaultValue={kudosMessage}
            onChange={(event) => {
              setKudosMessage(event.target.value);
            }}
            rows={5}
            style={{ marginTop: "2rem" }}
            variant="outlined"
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={() => setEditDialogOpen(false)}>
            Cancel
          </Button>
          <Button onClick={updateKudosCallback}
                  disabled={kudosMessage.trim().length == 0 ||
                            kudosRecipientMembers.length == 0}
                  autoFocus>
            Save
          </Button>
        </DialogActions>
      </Dialog>
      <Paper className="kudos-card">
        <div
          className="kudos-card-header"
          onClick={() => setExpanded(!expanded)}
        >
          <div className="members-container">
            {getRecipientComponent()}
            <Typography variant="body1">received kudos from</Typography>
            <Chip
              avatar={<Avatar src={getAvatarURL(sender?.workEmail)} />}
              label={sender?.name}
            />
          </div>
          <div className="kudos-status-container">{getStatusComponent()}</div>
        </div>
        <Divider />
        <Collapse in={expanded}>
          <div className="kudos-card-content">
            <Typography variant="body1">{kudos.message}</Typography>
            {kudos.recipientMembers?.length > 1 && (
              <div className="kudos-recipient-list">
                <Typography variant="body2">
                  {kudos.recipientTeam ? "Team Members:" : "Members:"}
                </Typography>
                {kudos.recipientMembers.map((recipient) => (
                  <Chip
                    key={recipient.id}
                    avatar={<Avatar src={getAvatarURL(recipient?.workEmail)} />}
                    label={`${recipient.firstName} ${recipient.lastName}`}
                  />
                ))}
              </div>
            )}
          </div>
        </Collapse>
      </Paper>
    </>
  );
};

KudosCard.propTypes = propTypes;

export default KudosCard;
