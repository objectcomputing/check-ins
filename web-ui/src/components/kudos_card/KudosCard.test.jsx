import React from 'react';
import KudosCard from './KudosCard';
import { AppContextProvider } from '../../context/AppContext';

const initialState = {
  state: {
    csrf: 'O_3eLX2-e05qpS_yOeg1ZVAs9nDhspEi',
    teams: [],
    userProfile: {
      id: "1",
      firstName: 'Jimmy',
      lastName: 'Johnson',
      role: ['MEMBER'],
    },
    terminatedMembers: [
      {
        id: "5",
        firstName: 'Jerry',
        lastName: 'Garcia',
        name: 'Jerry Garcia',
        role: ['MEMBER'],
      },
    ],
    memberProfiles: [
      {
        id: "1",
        firstName: 'Jimmy',
        lastName: 'Johnson',
        name: 'Jimmy Johnson',
        role: ['MEMBER'],
      },
      {
        id: "2",
        firstName: 'Jimmy',
        lastName: 'Olsen',
        name: 'Jimmy Olsen',
        role: ['MEMBER'],
      },
      {
        id: "3",
        firstName: 'Clark',
        lastName: 'Kent',
        name: 'Clark Kent',
        role: ['MEMBER'],
      },
      {
        id: "4",
        firstName: 'Kent',
        lastName: 'Brockman',
        name: 'Kent Brockman',
        role: ['MEMBER'],
      },
      {
        id: "6",
        firstName: 'Brock',
        lastName: 'Smith',
        name: 'Brock Smith',
        role: ['MEMBER'],
      },
      {
        id: "7",
        firstName: 'Jimmy',
        middleName: 'T.',
        lastName: 'Olsen',
        name: 'Jimmy T. Olsen',
        role: ['MEMBER'],
      },
    ],
  }
};

const terminated = {
  id: 'test-terminated-kudos',
  message: "Brock and Brockman did a great job helping Clark, Jimmy Olsen, Jimmy T. Olsen, and Johnson",
  senderId: "5",
  dateCreated: [ 2025, 2, 14 ],
  recipientMembers: [
      {
        id: "1",
        firstName: 'Jimmy',
        lastName: 'Johnson',
        role: ['MEMBER'],
      },
      {
        id: "2",
        firstName: 'Jimmy',
        lastName: 'Olsen',
        role: ['MEMBER'],
      },
      {
        id: "3",
        firstName: 'Clark',
        lastName: 'Kent',
        role: ['MEMBER'],
      },
      {
        id: "6",
        firstName: 'Brock',
        lastName: 'Smith',
        role: ['MEMBER'],
      },
      {
        id: "4",
        firstName: 'Kent',
        lastName: 'Brockman',
        role: ['MEMBER'],
      },
      {
        id: "7",
        firstName: 'Jimmy',
        middleName: 'T.',
        lastName: 'Olsen',
        role: ['MEMBER'],
      },
  ],
};

const kudos = {
  id: 'test-kudos',
  message: "Jimmy is awesome!",
  senderId: "1",
  dateCreated: [ 2025, 2, 17 ],
  recipientMembers: [
      {
        id: "2",
        firstName: 'Jimmy',
        lastName: 'Olsen',
        role: ['MEMBER'],
      },
  ],
};

it('inactive renders correctly', () => {
  snapshot(
    <AppContextProvider value={initialState}>
      <KudosCard
        kudos={terminated}
        includeActions
        includeEdit
        onKudosAction={() =>{}}
      />
    </AppContextProvider>
  );
});

it('active renders correctly', () => {
  snapshot(
    <AppContextProvider value={initialState}>
      <KudosCard
        kudos={kudos}
        includeActions
        includeEdit
        onKudosAction={() =>{}}
      />
    </AppContextProvider>
  );
});
