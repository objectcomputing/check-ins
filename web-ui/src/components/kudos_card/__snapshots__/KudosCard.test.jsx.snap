// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`active renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 kudos-card css-1ps6pg7-MuiPaper-root"
  >
    <div
      class="kudos-card-header"
    >
      <div
        class="members-container"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Jimmy Olsen
          </span>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          received kudos from
        </p>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Jimmy Johnson
          </span>
        </div>
      </div>
      <div
        class="kudos-status-container"
      >
        <div
          class="kudos-action-buttons"
        >
          <button
            aria-label="Approve"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorSuccess MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorSuccess css-1k9gzqu-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Edit"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EditIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Delete"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorError MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorError css-1ir7th6-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="kudos-card-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Jimmy is awesome!
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`inactive renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 kudos-card css-1ps6pg7-MuiPaper-root"
  >
    <div
      class="kudos-card-header"
    >
      <div
        class="members-container"
      >
        <div
          class="MuiAvatarGroup-root css-1ytufz-MuiAvatarGroup-root"
        >
          <div
            aria-label="Jimmy Olsen"
            class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar css-1wlk0hk-MuiAvatar-root"
            data-mui-internal-clone-element="true"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <div
            aria-label="Kent Brockman"
            class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar css-1wlk0hk-MuiAvatar-root"
            data-mui-internal-clone-element="true"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <div
            aria-label="Brock Smith"
            class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar css-1wlk0hk-MuiAvatar-root"
            data-mui-internal-clone-element="true"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <div
            aria-label="Clark Kent"
            class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar css-1wlk0hk-MuiAvatar-root"
            data-mui-internal-clone-element="true"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <div
            aria-label="Jimmy Olsen"
            class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar css-1wlk0hk-MuiAvatar-root"
            data-mui-internal-clone-element="true"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <div
            aria-label="Jimmy Johnson"
            class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar css-1wlk0hk-MuiAvatar-root"
            data-mui-internal-clone-element="true"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          received kudos from
        </p>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="http://localhost:8080/services/member-profiles/member-photos/undefined"
            />
          </div>
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            Jerry Garcia
          </span>
        </div>
      </div>
      <div
        class="kudos-status-container"
      >
        <div
          class="kudos-action-buttons"
        >
          <button
            aria-label="Approve"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorSuccess MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorSuccess css-1k9gzqu-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Edit"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EditIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Delete"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorError MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorError css-1ir7th6-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="kudos-card-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Brock and Brockman did a great job helping Clark, Jimmy Olsen, Jimmy T. Olsen, and Johnson
            </p>
            <div
              class="kudos-recipient-list"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                Members:
              </p>
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Jimmy Johnson
                </span>
              </div>
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Jimmy Olsen
                </span>
              </div>
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Clark Kent
                </span>
              </div>
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Brock Smith
                </span>
              </div>
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Kent Brockman
                </span>
              </div>
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiChip-avatar MuiChip-avatarMedium MuiChip-avatarColorDefault css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  Jimmy Olsen
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
